src/main/java/duke/Command/ByC.java

```diff
@@ -0,0 +1,41 @@
+package duke.Command;
+
+import duke.*;
+import duke.Command.Command;
+import duke.task.Deadline;
+import duke.task.Task;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class ByC extends Command {
+
+    private final String input;
+
+    public ByC(String input) {
+        this.input = input;
+    }
+
+    @Override
+    public String execute(Ui ui, TaskList todoList, Storage store) throws IOException {
+        LocalDate checkBy = LocalDate.parse(input.substring(3));
+        String result = "";
+        result += "By this day, you have: \n" ;
+        int deadCount = 0;
+        for (Task checkDead : todoList.todoList) {
+            if (checkDead instanceof Deadline &&
+                    (((Deadline) checkDead).by.isBefore(checkBy) ||
+                            ((Deadline) checkDead).by.equals(checkBy))) {
```

Comment:
This seems like a complicated expression.

Perhaps it would be clearer to separate it? It will improve code quality!

https://nus-cs2103-ay2021s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions Here's some advice!

Reply:
fantastic points thuya, thank you!!